src/status_im/contexts/settings/wallet/keypairs_and_accounts/actions/view.cljs

```diff
@@ -1,44 +1,53 @@
 (ns status-im.contexts.settings.wallet.keypairs-and-accounts.actions.view
   (:require [quo.core :as quo]
+            [react-native.core :as rn]
             [status-im.contexts.settings.wallet.keypairs-and-accounts.remove.view :as remove-key-pair]
             [utils.i18n :as i18n]
             [utils.re-frame :as rf]))
 
-(defn on-rename-key-pair
-  [key-pair]
-  (rf/dispatch [:open-modal :screen/settings.rename-keypair key-pair]))
-
-(defn on-remove-key-pair
-  [key-pair]
-  (rf/dispatch [:show-bottom-sheet
-                {:theme   :dark
-                 :content (fn []
-                            [remove-key-pair/view key-pair])}]))
-
-(defn on-show-qr
-  [data]
-  (rf/dispatch [:open-modal :screen/settings.encrypted-key-pair-qr data]))
-
 (defn view
-  [props key-pair]
-  (let [has-paired-device (rf/sub [:pairing/has-paired-devices])]
+  [props keypair]
+  (let [has-paired-device (rf/sub [:pairing/has-paired-devices])
+        missing-keypair?  (rn/use-memo #(= (:stored props) :missing) [props])
+        on-scan-qr        (rn/use-callback #(rf/dispatch [:open-modal :screen/settings.scan-keypair-qr
+                                                          [(:key-uid keypair)]])
+                                           [keypair])
+        on-show-qr        (rn/use-callback #(rf/dispatch [:open-modal
+                                                          :screen/settings.encrypted-key-pair-qr
+                                                          keypair])
+                                           [keypair])
+        on-remove-keypair (rn/use-callback #(rf/dispatch
+                                             [:show-bottom-sheet
+                                              {:theme   :dark
+                                               :content (fn []
+                                                          [remove-key-pair/view keypair])}])
+                                           [keypair])
+        on-rename-keypair (rn/use-callback #(rf/dispatch [:open-modal :screen/settings.rename-keypair
+                                                          keypair])
+                                           [keypair])]
     [:<>
      [quo/drawer-top props]
      [quo/action-drawer
       [(when has-paired-device
-         [{:icon                :i/qr-code
-           :accessibility-label :show-key-pr-qr
-           :label               (i18n/label :t/show-encrypted-qr-of-key-pairs)
-           :on-press            #(on-show-qr key-pair)}])
+         (concat
+          (if-not missing-keypair?
```

Comment:
Loved that you are handling the edge case if the keypair is missing, we shouldn't show this option 🙌